File: src/NetWorthCalculator.Web.ReactJs/src/Models/Liability/Liability.ts
import { observable } from "mobx";
import { serializable } from "serializr";
import { LiabilityGroup } from "./LiabilityGroup";
import { PaymentInterval } from "./PaymentInterval";

interface ILiability {
	id: number;
	amount: number;
	description: string;
	group: LiabilityGroup;
	intervalAmount: number;
	order: number;
	paymentInterval: PaymentInterval;
}

export default class Liability {
	@serializable @observable id!: number;

	@serializable @observable amount!: number;

	@serializable @observable description!: string;

	@serializable @observable group!: LiabilityGroup;

	@serializable @observable intervalAmount!: number;

	@serializable @observable order!: number;

	@serializable @observable paymentInterval!: PaymentInterval;

	constructor(values: ILiability) {
		Object.assign(this, { ...values });
	}
}

File: src/NetWorthCalculator.Entities/Models/Liability.cs
using NetWorthCalculator.Entities.Enums;

namespace NetWorthCalculator.Entities.Models
{
    public class Liability
    {
        public int Id { get; set; }

        public decimal Amount { get; set; }

        public string Description { get; set; }

        public LiabilityGroup Group { get; set; }

        public decimal IntervalAmount { get; set; }

        public int Order { get; set; }

        public PaymentInterval PaymentInterval { get; set; }
    }
}
